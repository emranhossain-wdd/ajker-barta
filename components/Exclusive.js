import { formatDistanceToNow } from 'date-fns';
import { useRouter } from 'next/router';
import React, { useState } from 'react'
import styles from '../styles/exclusive.module.css'
const Exclusive = ({ newses }) => {
  const latest = newses?.reverse()
  const router = useRouter()
  const [isLetest, setIsLetest] = useState(true)
  let count = 1
  return (
    <div data-testid='exclusiveId' className='pb-5'>
      <div className="container">
        <div className="grid lg:grid-cols-2 gap-3">
          <div>
            <button onClick={() => setIsLetest(true)} className={isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Letest</button>
            <button onClick={() => setIsLetest(false)} className={!isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Trending</button>
            {
              isLetest ? <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(0, 8).map(news => <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div> : <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(8, 16).map(news => <div onClick={() => router.push(`/news/${news._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div>
            }

          </div>
          <div>
            <h1 className={styles.exclusive}>Satireday</h1>
            <div className='grid sm:grid-cols-2 gap-3'>
              {
                latest?.slice(16, 20).map(news => <div className='cursor-pointer imageEffect' onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} key={news?._id}>
                  <div className='overflow-hidden'>
                    <img className='w-full h-32 object-cover' src={news?.images.img1} alt="" />
                  </div>
                  <h1 className='text-lg my-1 font-medium leading-5 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                  <p className='text-sm leading-4'>{news?.description[0]?.slice(0, 100)}</p>
                  <p className="px-2 mt-2 py-1 font-medium text-sm rounded-full bg-gray-100 w-fit text-blue-500">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</p>
                </div>)
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Exclusive;