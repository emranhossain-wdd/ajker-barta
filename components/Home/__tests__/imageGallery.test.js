import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ImageGallery from '../imageGallery/ImageGallery'
import { render } from '@testing-library/react'
import { testArr } from '../../__tests__/Exclusive.test'

it('render imageGallery component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<ImageGallery />, div)
})

it('testing imageGallery with props', () => {
  const { getByTestId } = render(<ImageGallery testArr={testArr} />)
  expect(getByTestId('imageGalleryId')).toBeTruthy()
})
it('maching snapshot', () => {
  const tree = renderer.create(<ImageGallery />)
  expect(tree).toMatchSnapshot()
})