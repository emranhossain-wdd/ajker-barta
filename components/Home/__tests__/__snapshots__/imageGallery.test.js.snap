// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="pb-5"
  data-testid="exclusiveId"
>
  <div
    className="container"
  >
    <div
      className="grid lg:grid-cols-2 gap-3"
    >
      <div>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white"
          onClick={[Function]}
        >
          Letest
        </button>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2"
          onClick={[Function]}
        >
          Trending
        </button>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
      <div>
        <h1
          className="exclusive"
        >
          Satireday
        </h1>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`maching snapshot 2`] = `
<div
  className="container overflow-hidden "
  data-testid="imageGalleryId"
>
  <h2
    className="text-xl text-blue-900 font-semibold mb-3 mt-5"
  >
    <span
      className="text-gray-700"
    >
       Todays Gallery 
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-red-500 css-i4bv87-MuiSvgIcon-root"
      data-testid="ChevronRightIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </h2>
  <div
    className="galleryRow"
  >
    <div
      className="gallery"
    >
      <img
        alt=""
        className="w-full h-full object-cover"
        src="https://i.ibb.co/1zDMSrv/bird-2.jpg"
      />
      <div>
        <h1
          className="text-lg leading-5 font-semibold absolute bottom-4 left-3 text-white"
        >
          “I’m Like A Bird” by Nelly Furtado
        </h1>
        <svg
          className="absolute top-2 left-2 text-5xl"
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41,42H13c-2.2,0-4-1.8-4-4V18c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C45,40.2,43.2,42,41,42z"
            fill="#E65100"
          />
          <path
            d="M35,36H7c-2.2,0-4-1.8-4-4V12c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C39,34.2,37.2,36,35,36z"
            fill="#F57C00"
          />
          <circle
            cx="30"
            cy="16"
            fill="#FFF9C4"
            r="3"
          />
          <polygon
            fill="#942A09"
            points="17,17.9 8,31 26,31"
          />
          <polygon
            fill="#BF360C"
            points="28,23.5 22,31 34,31"
          />
        </svg>
      </div>
    </div>
    <div
      className="gallery"
    >
      <img
        alt=""
        className="w-full h-full object-cover"
        src="https://i.ibb.co/D1dqj8R/girl.jpg"
      />
      <div>
        <h1
          className="text-lg leading-5 font-semibold absolute bottom-4 left-3 text-white"
        >
          Smiling Girl
        </h1>
        <svg
          className="absolute top-2 left-2 text-5xl"
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41,42H13c-2.2,0-4-1.8-4-4V18c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C45,40.2,43.2,42,41,42z"
            fill="#E65100"
          />
          <path
            d="M35,36H7c-2.2,0-4-1.8-4-4V12c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C39,34.2,37.2,36,35,36z"
            fill="#F57C00"
          />
          <circle
            cx="30"
            cy="16"
            fill="#FFF9C4"
            r="3"
          />
          <polygon
            fill="#942A09"
            points="17,17.9 8,31 26,31"
          />
          <polygon
            fill="#BF360C"
            points="28,23.5 22,31 34,31"
          />
        </svg>
      </div>
    </div>
    <div
      className="gallery"
    >
      <img
        alt=""
        className="w-full h-full object-cover"
        src="https://i.ibb.co/L16Dbqr/family.jpg"
      />
      <div>
        <h1
          className="text-lg leading-5 font-semibold absolute bottom-4 left-3 text-white"
        >
          Happy Family
        </h1>
        <svg
          className="absolute top-2 left-2 text-5xl"
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41,42H13c-2.2,0-4-1.8-4-4V18c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C45,40.2,43.2,42,41,42z"
            fill="#E65100"
          />
          <path
            d="M35,36H7c-2.2,0-4-1.8-4-4V12c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C39,34.2,37.2,36,35,36z"
            fill="#F57C00"
          />
          <circle
            cx="30"
            cy="16"
            fill="#FFF9C4"
            r="3"
          />
          <polygon
            fill="#942A09"
            points="17,17.9 8,31 26,31"
          />
          <polygon
            fill="#BF360C"
            points="28,23.5 22,31 34,31"
          />
        </svg>
      </div>
    </div>
    <div
      className="gallery"
    >
      <img
        alt=""
        className="w-full h-full object-cover"
        src="https://i.ibb.co/9nNLBv0/millionear.jpg"
      />
      <div>
        <h1
          className="text-lg leading-5 font-semibold absolute bottom-4 left-3 text-white"
        >
          Success
        </h1>
        <svg
          className="absolute top-2 left-2 text-5xl"
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41,42H13c-2.2,0-4-1.8-4-4V18c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C45,40.2,43.2,42,41,42z"
            fill="#E65100"
          />
          <path
            d="M35,36H7c-2.2,0-4-1.8-4-4V12c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4v20C39,34.2,37.2,36,35,36z"
            fill="#F57C00"
          />
          <circle
            cx="30"
            cy="16"
            fill="#FFF9C4"
            r="3"
          />
          <polygon
            fill="#942A09"
            points="17,17.9 8,31 26,31"
          />
          <polygon
            fill="#BF360C"
            points="28,23.5 22,31 34,31"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
